api/src/data_migration/transformation/subtask/transform_competition.py

```diff
@@ -41,16 +42,25 @@ def transform_records(self) -> None:
 
                 if opportunity_cfda:
                     opportunity_id = opportunity_cfda.opportunity_id
-                    opportunity_assistance_listing_id = opportunity_cfda.opp_cfda_id
 
                     # Make sure the opportunity exists in our target table
                     opportunity = self.db_session.execute(
-                        select(Opportunity).where(Opportunity.opportunity_id == opportunity_id)
+                        select(Opportunity)
+                        .where(Opportunity.opportunity_id == opportunity_cfda.opportunity_id)
+                        .options(selectinload(Opportunity.opportunity_assistance_listings))
                     ).scalar_one_or_none()
 
-                    if not opportunity:
-                        opportunity_id = None
-                        opportunity_assistance_listing_id = None
+                    if opportunity:
+                        for assistance_listing in opportunity.opportunity_assistance_listings:
+                            if (
+                                assistance_listing.opportunity_assistance_listing_id
+                                == opportunity_cfda.opp_cfda_id
+                            ):
+                                opportunity_id = opportunity.opportunity_id
+                                opportunity_assistance_listing_id = (
+                                    opportunity_assistance_listing_id
+                                )
```

Comment:
I don't think what you have right now sets the value right.
```suggestion
                                opportunity_assistance_listing_id = (
                                    assistance_listing.opportunity_assistance_listing_id
                                )
```

Reply:
updated